internal/controller/dynamicReconciler.go

```diff
@@ -0,0 +1,131 @@
+package controller
+
+import (
+	"context"
+	kumquatv1beta1 "kumquat/api/v1beta1"
+	"kumquat/repository"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type DynamicReconciler struct {
+	client.Client
+	GVK          schema.GroupVersionKind
+	K8sClient    K8sClient
```

Comment:
I keep seeing both `K8sClient` and `Client` getting used; how do we make the determination of which one to use? Is it actually needed to have both?